Plugins/Detray/src/DetrayConverter.cpp

```diff
@@ -0,0 +1,324 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Plugins/Detray/DetrayConverter.hpp"
+
+#include "Acts/Detector/Detector.hpp"
+#include "Acts/Detector/DetectorVolume.hpp"
+#include "Acts/Detector/Portal.hpp"
+#include "Acts/Navigation/PortalNavigation.hpp"
+#include "Acts/Plugins/Detray/DetrayConversionHelper.hpp"
+#include "Acts/Surfaces/CylinderBounds.hpp"
+#include "Acts/Surfaces/CylinderSurface.hpp"
+#include "Acts/Surfaces/DiscSurface.hpp"
+#include "Acts/Surfaces/RadialBounds.hpp"
+#include "Acts/Surfaces/RegularSurface.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+#include "Acts/Surfaces/SurfaceBounds.hpp"
+
+#include "detray/io/frontend/detector_writer.hpp"
+
+namespace {
+
+/// Find the position of the volume to point to
+///
+/// @param volume the volume to find
+/// @param the collection of volumes
+///
+/// @note return -1 if not found, to be interpreted by the caller
+int findVolume(
+    const Acts::Experimental::DetectorVolume* volume,
+    const std::vector<const Acts::Experimental::DetectorVolume*>& volumes) {
+  auto candidate = std::find(volumes.begin(), volumes.end(), volume);
+  if (candidate != volumes.end()) {
+    return std::distance(volumes.begin(), candidate);
+  }
+  return -1;
+}
+}  // namespace
+
+/// detray geometry writer function, debug purposes
+void Acts::DetrayConverter::writeToJson(
+    const DetrayDetector& dDetector,
+    const typename DetrayDetector::name_map& names) {
+  auto writer_cfg = detray::io::detector_writer_config{}
+                        .format(detray::io::format::json)
+                        .replace_files(true);
```

Comment:
Can we maybe pass the writer config as an argument, so that the IO can be configure from outside of this function? E.g.
```suggestion
    const typename DetrayDetector::name_map& names,
    detray::io::detector_writer_config writer_cfg = {}) {
    // This method writes a detector to json format
    writer_cfg.format(detray::io::format::json);
```